### DISCUSSION TITLE
Limit the number of upgrades per PR

### DISCUSSION THREAD

**Discussion by toestreich** [renovatebot/renovate#21804] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

Right now we have a package rule set up for all non-major dependencies. This has become an issue for repos that are newly onboarded to renovate since they have a bunch of out of date packages and a single PR with all of these small changes often times results in failed builds. 

We're looking at a way to divide and conquer these larger PRs so in the event where a breaking change is introduced, it isn't preventing other upgrades from being merged and it is also easier to identify which package is the problem. 

Does renovate currently have a way to limit the number of packages being upgraded in a single PR? If not, I think this could be a great feature add to teams who have a huge number of out of date packages

**Comment by viceice** [comment]
don't group all updates until the repo is nearly up to date?

**Comment by HonkingGoose** [comment]
Did you read our [Noise Reduction docs](https://docs.renovatebot.com/noise-reduction/)? It lists multiple strategies to reduce the noise of Renovate updates:

- Package grouping, and how to do this smartly
- Scheduling when Renovate runs
- Automerging
- Lock file considerations

Grouping too many dependencies is "asking for problems". I avoid grouping things manually, the default grouping is good for me. I like having small actionable PRs from Renovate. It may feel "noisy", but it's quicker than figuring out why a ball of 20 dependencies is tangled up.

By the way, if you're on GitHub, and want to merge Renovate PRs faster, check out [our Merge Queue support]( https://docs.renovatebot.com/key-concepts/automerge/#github-merge-queue).

**Reply by toestreich** [reply]
Yea we're trying to avoid grouping too many, thats why I was hoping there was a built in feature that would prevent this like limiting the number of packages upgraded per PR. Getting all repos up to speed upgrading individual packages would take a long time since we have some with a very very large amount of upgrades to do :sweat_smile: 

The merge queue seems interesting, Ill take a deeper look, thanks!
